backend/layers/persistence/persistence.py

```diff
@@ -605,6 +654,22 @@ def get_dataset_version(self, dataset_version_id: DatasetVersionId, get_tombston
                     return None
             return self._hydrate_dataset_version(dataset_version)
 
+    def get_all_dataset_versions_for_collection(
+        self, collection_id: CollectionId, from_date: datetime = datetime.min
```

Comment:
is `datetime.min` immutable? If it's not then change it to:

 `from_date: Optional[datetime] = None`

and add this

`from_date = from_date if from_date else datetime.min`

This is to avoid weird bugs related to modifying a mutable attribute.

Reply:
datetime.min is a constant attribute: the earliest possible `datetime` value. I don't think this changes the extent to which this is bug-prone but I'll change the logic because I agree, better to avoid `not` statements where possible.

Reply:
done ✔️